### PR TITLE
Send blockdev change signal on grow physical

### LINKED ISSUE #3371: pool.GrowPhysicalDevice doesn't trigger change notification for blockdev.TotalPhysicalSize
Subject says it all. :-)

### PR DESCRIPTION
**Pull request by mulkieran**
Closes #3371

### CODE REVIEW

File: src/dbus_api/tree.rs [hunk 1]
```diff
@@ -857,6 +860,47 @@ impl DbusTreeHandler {
...
+                    Vec::new(),
+                    consts::BLOCKDEV_TOTAL_SIZE_PROP.to_string() =>
+                    box_variant!(total_physical_size_prop.clone())
+                },
+                consts::BLOCKDEV_INTERFACE_NAME_3_6 => {
+                    Vec::new(),
+                    consts::BLOCKDEV_TOTAL_SIZE_PROP.to_string() =>
+                    box_variant!(total_physical_size_prop)
+                }
+            ),
```
Comment (jbaublitz) [inline src/dbus_api/tree.rs:895]: I'm not sure this is the best thing to do here. Even if grow physical was added in r3, the total physical size property is available all the way back so theoretically even if a user is using a r3 method, we should notify them that the property in r1 (for example) has changed.

**Review by jbaublitz — CHANGES_REQUESTED** [review]

File: src/dbus_api/tree.rs (hunk 1 above)
Comment (mulkieran) [reply src/dbus_api/tree.rs:895]: I agree.

**Comment by mulkieran** [comment]
I think that the best way to get the object path corresponding to the device is to relinquish the Some-Write lock and get an AllRead lock.

**Review by bgurney-rh — APPROVED** [review]

**Comment by bgurney-rh** [comment]
The new stratisd_cert test that was added today has a --monitor-dbus failure for this DifferentProperty message:
```
DifferentProperty(dbus.ObjectPath('/org/storage/stratis3/13'), dbus.String('org.storage.stratis3.blockdev.r0'), dbus.String('UserInfo'), dbus.Struct((dbus.Boolean(False), dbus.String('')), signature=None, variant_level=1), dbus.Struct((dbus.Boolean(True), dbus.String('eeeeeeeeee')), signature=None, variant_level=1))
```

**Comment by mulkieran** [comment]
rebased

**Comment by mulkieran** [comment]
> The new stratisd_cert test that was added today has a --monitor-dbus failure for this DifferentProperty message:
> 
> ```
> DifferentProperty(dbus.ObjectPath('/org/storage/stratis3/13'), dbus.String('org.storage.stratis3.blockdev.r0'), dbus.String('UserInfo'), dbus.Struct((dbus.Boolean(False), dbus.String('')), signature=None, variant_level=1), dbus.Struct((dbus.Boolean(True), dbus.String('eeeeeeeeee')), signature=None, variant_level=1))
> ```

CI should be running again w/  #3375

**Comment by mulkieran** [comment]
rebased.

**Review by jbaublitz — APPROVED** [review]

**Comment by mulkieran** [comment]
@mvollmer Please let us know if you want to test this.

**Comment by mulkieran** [comment]
rebased + squashed

**Comment by mulkieran** [comment]
Now all new and revised!

**Review by jbaublitz — APPROVED** [review]

**Comment by mulkieran** [comment]
Squashing a bit before merging...

**Pull request merged**